pwndbg/gdblib/symbol.py

```diff
@@ -184,40 +188,36 @@ def get(address, gdb_only=False):
 
 
 @pwndbg.lib.memoize.reset_on_objfile
-def address(symbol, allow_unmapped=False):
-    if isinstance(symbol, int):
-        return symbol
-
-    try:
-        return int(symbol, 0)
-    except Exception:
-        pass
-
+def address(symbol: str) -> int:
+    """
+    Get the address for `symbol`
+    """
     try:
         symbol_obj = gdb.lookup_symbol(symbol)[0]
         if symbol_obj:
             return int(symbol_obj.value().address)
-    except Exception:
-        pass
-
-    try:
-        result = gdb.execute("info address %s" % symbol, to_string=True, from_tty=False)
-        address = int(re.search("0x[0-9a-fA-F]+", result).group(), 0)
-
-        # The address found should lie in one of the memory maps
-        # There are cases when GDB shows offsets e.g.:
-        # pwndbg> info address tcache
-        # Symbol "tcache" is a thread-local variable at offset 0x40
-        # in the thread-local storage for `/lib/x86_64-linux-gnu/libc.so.6'.
-        if not allow_unmapped and not pwndbg.vmmap.find(address):
-            return None
-
-        return address
-
-    except gdb.error:
-        return None
+    except gdb.error as e:
+        # Symbol lookup only throws exceptions on errors, not if it failed to
+        # lookup a symbol. We want to raise these errors so we can handle them
+        # properly, but there are some we haven't figured out how to fix yet, so
+        # we ignore those here
+        skipped_exceptions = []
```

Comment:
Should be:

```py
skipped_exceptions = [
    # comment here
    "No frame selected",
    ...
]
```
but this can be changed in a next PR as well